src/Shared/TaskParameter.cs

```diff
@@ -284,11 +284,8 @@ private ITaskItem CreateNewTaskItemFrom(ITaskItem copyFrom)
                 escapedDefiningProject = copyFromAsITaskItem2.GetMetadataValueEscaped(FileUtilities.ItemSpecModifiers.DefiningProjectFullPath);
                 IDictionary nonGenericEscapedMetadata = copyFromAsITaskItem2.CloneCustomMetadataEscaped();
 
-                if (nonGenericEscapedMetadata is Dictionary<string, string>)
-                {
-                    escapedMetadata = (Dictionary<string, string>)nonGenericEscapedMetadata;
-                }
-                else
+                escapedMetadata = nonGenericEscapedMetadata as Dictionary<string, string>;
```

Comment:
Why this structure? I would've preferred more "obj is type name."

Reply:
It is because we need here to have a bigger visibility scope for `escapedMetadata` variable than `is` pattern allows and I did not want to create a new variable given that `as` pattern works well.